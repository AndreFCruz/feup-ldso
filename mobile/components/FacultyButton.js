import React from 'react';
import PropTypes from 'prop-types';

import {
    Image,
    View,
    TouchableOpacity
} from 'react-native';
import {heightPercentageToDP as hp} from 'react-native-responsive-screen';

import { logos } from '../constants/Logos';

class FacultyButton extends React.Component {
    render() {
        return (
            <TouchableOpacity title="" onPress={this.props.onPress}>
                <View style={styles['button']} >
                    <Image style={styles.image} source={logos[this.props.name].uri} />
                </View>
            </TouchableOpacity>
        );
    }
}

FacultyButton.propTypes = {
    onPress: PropTypes.func,
    name: PropTypes.string.isRequired
};

export { FacultyButton };

const styles = {
    
    image: {
        flex: 1,
        aspectRatio: 0.9,
        resizeMode: 'contain',
        transform: [{
            rotate: '315deg'
        }],
    },

    button: {
        height: hp('18%'),
        width: hp('18%'),
        borderWidth: 0,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'white',
        borderRadius: hp('6%'),
        margin: hp('1%'),
    }
};
