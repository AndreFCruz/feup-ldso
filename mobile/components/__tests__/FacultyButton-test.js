import 'react-native';
import React from 'react';
import { FacultyButton } from '../FacultyButton';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders correct logo', () => {
    const mockFunction = jest.fn();
    
    const wrapper = shallow(<FacultyButton name={'feup'} onPress={mockFunction} />);

    expect(wrapper).toMatchSnapshot();
});

it('calls onPress function', () => {
    const mockFunction = jest.fn();
    
    const wrapper = shallow(<FacultyButton name={'feup'} onPress={mockFunction} />);

    wrapper.find('TouchableOpacity').props().onPress();

    expect(mockFunction.mock.calls.length).toBe(1);
});