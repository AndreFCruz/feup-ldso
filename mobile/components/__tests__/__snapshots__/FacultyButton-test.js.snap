// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct logo 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  title=""
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 80,
        "borderWidth": 0,
        "height": 240,
        "justifyContent": "center",
        "margin": 13.5,
        "width": 240,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "aspectRatio": 0.9,
          "flex": 1,
          "resizeMode": "contain",
          "transform": Array [
            Object {
              "rotate": "315deg",
            },
          ],
        }
      }
    />
  </Component>
</TouchableOpacity>
`;
